pkg/provisioner/ironic/ironic.go

```diff
@@ -590,11 +591,19 @@ func setDeployImage(driverInfo map[string]interface{}, config ironicConfig, acce
 				return deployImageInfo
 			}
 		case metal3v1alpha1.ImageFormatInitRD:
-			if allowInitRD {
+			if hostImage.KernelURL != "" {
+				deployImageInfo[deployKernelKey] = hostImage.KernelURL
+			} else if config.deployKernelURL == "" {
+				return nil
```

Comment:
I don't think returning here is consistent with the previous behaviour when there is no kernel available. The code on ll. 610-614 should still have a chance to run.

Reply:
I don't think there is a point in it. If we reach this line, we have a request for initrd, but the kernel is neither in the image, nor in the configuration. There is no way we can succeed.

Even worse, we can end up with providing a pre-configured ISO when the image requests `ImageFormatInitRD`.

Reply:
It would be better to output  log info before returning.